contracts/VestingEscrow.vy

```diff
@@ -0,0 +1,369 @@
+# @version 0.3.7
+
+"""
+@title Vesting Escrow
+@author Curve Finance, Yearn Finance, Lido Finance
+@license MIT
+@notice Vests ERC20 tokens for a single address
+@dev Intended to be deployed many times via `VotingEscrowFactory`
+"""
+
+from vyper.interfaces import ERC20
+
+
+interface IVestingEscrowFactory:
+    def voting_adapter() -> address: nonpayable
+    def owner() -> address: nonpayable
+    def manager() -> address: nonpayable
+
+
+event VestingEscrowInitialized:
+    factory: indexed(address)
+    recipient: indexed(address)
+    token: indexed(address)
+    amount: uint256
+    start_time: uint256
+    end_time: uint256
+    cliff_length: uint256
+    is_fully_revokable: bool
+
+
+event Claim:
+    recipient: indexed(address)
+    beneficiary: indexed(address)
+    claimed: uint256
+
+
+event UnvestedTokensRevoked:
+    recoverer: indexed(address)
+    revoked: uint256
+
+
+event VestingFullyRevoked:
+    recoverer: indexed(address)
+    revoked: uint256
+
+
+event ERC20Recovered:
+    token: address
+    amount: uint256
+
+
+event ETHRecovered:
+    amount: uint256
+
+
+recipient: public(address)
+token: public(ERC20)
+start_time: public(uint256)
+end_time: public(uint256)
+cliff_length: public(uint256)
+factory: public(IVestingEscrowFactory)
+total_locked: public(uint256)
+is_fully_revokable: public(bool)
+
+total_claimed: public(uint256)
+disabled_at: public(uint256)
```

Comment:
can we have a doc about all public fields?

Reply:
Added to spec https://hackmd.io/FZEW9MM7QwmUehFtKDmqnw?view#Public-variables1